### PR TITLE
Replace HTML newline earlier on in logic

### PR DESCRIPTION
**Pull request by AddisonDunn**
## Product Description

Note: time-sensitive, we are trying to get this out with the ~~Sunday night~~ Wednesday deploy.

In response to [this ticket](https://dimagi-dev.atlassian.net/browse/USH-3373?atlOrigin=eyJpIjoiNTI0ZWI1YjMyMWQxNDczMDgxY2Q1NjVmODY1NzRiY2YiLCJwIjoiaiJ9).

Header and bullet markdown isn't working in case tiles. The small change in this PR makes both of these work.

## Technical Summary

The issue with header markdown stemmed from markdown-it's processing of the HTML newline character: `&#10;`. As described in the ticket, it doesn't do the header markdown processing properly for that character, but it does for `\n`. Notice how the header incorrectly continues on with `&#10;`, but not for `\n`:  

```
md.render("## Name ## &#10;Please call the contact...")
> '<h2><strong>Name</strong> ## \nPlease call the contact...</h2>\n'
md.render("## Name ## \nPlease call the contact...")
> '<h2><strong>Name</strong></h2>\n<p>Please call the contact...</p>\n'
```

I believe the issue with bullet markdown stemmed from this same issue as well. Bullet markdown seems to work properly under this change.

Per Jenny's description in #33155, the markdown library changes `&#10;`s to `\n`s anyways, so simply replacing the `\n`s in the HTML before it's rendered should be a safe way to get markdown-it to process the markdown as intended.

### Safety story

The logic of this is quite sound, I believe (See the last paragraph in the section above). But I've tested locally so far, and am open to other ideas on how to make this feel safer.

We can also consider limiting this change to case tiles somehow, to limit the possible scope.

### Automated test coverage

No.

### QA Plan

Unfortunately, I don't think there's time before Sunday night.

### Rollback instructions

- [X] This PR can be reverted after deploy with no further considerations

^^ There's one non-technical caveat, which is that USH delivery should be informed beforehand, so that they know to expect that their display property calcs will be shown in the UI differently.

### Labels & Review
- [X] Risk label is set correctly
- [X] The set of people pinged as reviewers is appropriate for the level of risk of the change

### CODE REVIEW

**Comment by nospame** [comment]
> Adding to the risk here: there is a caveat to this change, even though we're perhaps working more inline with the way markdown is supposed to work. See [here](https://dimagi-dev.atlassian.net/browse/USH-3373?focusedCommentId=274130). But if we think it's unlikely other projects are using multiple `&#10;`s in a row it might not be a worry. Or maybe we can limit this to case tiles to minimize any downside.

@AddisonDunn I haven't tested it, but it looks like you should be able to set markdownit to explicitly recognize `\n` as line breaks. https://github.com/markdown-it/markdown-it/blob/2b6cac25823af011ff3bc7628bc9b06e483c5a08/README.md?plain=1#L110

I think usage would look like `const md = window.markdownit().set({ breaks: true });`

File: corehq/apps/cloudcare/static/cloudcare/js/utils.js [hunk 1]
```diff
@@ -233,8 +233,8 @@ hqDefine('cloudcare/js/utils', [
     };
 
     function renderMarkdown(text) {
-        const md = window.markdownit();
-        return md.render(DOMPurify.sanitize(text || "")).replaceAll("\n", "<br>");
+        const md = window.markdownit().set({ breaks: true });
```
Comment (nospame) [inline corehq/apps/cloudcare/static/cloudcare/js/utils.js:236]: nit (even though this is the syntax I originally suggested):
```suggestion
        const md = window.markdownit({ breaks: true });
```

**Review by nospame — APPROVED** [review]
This seems pretty straightforward and reasonable to me.

**Review by MartinRiese — APPROVED** [review]

**Comment by orangejenny** [comment]
The change to replace `&#10;` before rendering markdown instead of after seems good and also low-risk, since it fixes a bug that was just introduced a few days ago.

I'm not clear on the rationale for the markdown setting change. It doesn't look like it's part of the bug fix, or is it?

**Comment by orangejenny** [comment]
@AddisonDunn aha, thank you. Hmm, this is a little tricky. It's very compelling that adding `breaks` adds support for functionality that currently can't be accomplished any other way. It's unfortunate that figuring out which projects might be affected would mean parsing all apps case list/detail config and forms looking for markdown with multiple line breaks, even though the number of affected non-USH projects is likely to be small, since this is a web apps only change.

Limiting to case tiles would indeed reduce the scope of this change, but markdown ought to be rendered consistently across the product (at least across web apps, there may be discrepancies on mobile due to different markdown libraries), even if that increases the number of projects that might see a behavior change here.

Agree it would be good to inform delivery beforehand. Can you just check in with Cody or Kim that they're at peace with just announcing this change internally and not doing other investigation/outreach?

**Comment by orangejenny** [comment]
> Though I'm not familiar with form markdown, so I don't know if its the same there.

Same, I don't know offhand if it'll work the same way in forms, and if it doesn't, there could be a fair number of projects affected...although, since it's web apps only, the scope still probably isn't that large.

Weekly deploy sounds reasonable, looks like delivery was already notified.

**Pull request merged**